### PR TITLE
new posting

### PR DESCRIPTION
**Pull request by wachoi**
@pawsong I wrote a new posting about Material UI. Review this plz.

### CODE REVIEW

**Comment by wachoi** [comment]
@pawsong I added npm trends image as you commented. Check it plz! Plus, what do you think is better? Putting Material library's 2014 readme in Quote or Code?

**Comment by wachoi** [comment]
@pawsong I wrote a new posting about Material UI theme and styles. Check it please when you have enough time.

File: posts/8_materialUI2.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+그 다음은 `@material-ui/styles` 또는 `@material-ui/core/styles`에서 제공하는 makeStyles, styled, withStyles를 이용하는 방법이다. 각각의 사용법은 [이 문서](https://material-ui.com/styles/basics/#material-ui-styles)에 자세히 나와 있다. Theme은 여러 component들이 공유하는 속성들을 한꺼번에 바꾸는데 사용하는데, makeStyles 등은 그보다 좀더 섬세하게 하나 하나 스타일을 바꾸고자 할 때 사용하는 것 같다. 사실 theme에서도 내가 위에서 한 것처럼 특정한 component의 스타일을 바꿀 수 있다. 아마 여기서 다루는 API들도 내가 한 것과 비슷한 방식으로 특정한 상황에서만 수정한 theme이 사용되도록 하는게 아닌가 싶다.
+
+솔직히 이게 어떻게 작동하는 건지 잘 이해를 못했다. 지금 이해한대로만 간단히 언급하고, React의 기본 개념들을 좀더 공부한 다음에 다시 살펴봐야겠다.
+
+1. **Hook API**: makeStyles를 이용해 style sheet를 만들고, className을 이용해 특정한 component에 적용시키는 방법. CSS selector로 특정한 html tag에 스타일을 적용하는 것과 비슷한 것 같다. 한 component를 여러 스타일로 써야 하는데, 전체 코드가 길지 않아서 스타일을 나타내는 코드와 그 스타일을 사용하는 component가 있는 코드를 같이 보는 게 가독성이 더 좋을 때 사용하면 좋을 것 같다.
+
+2. **Styled components API**: styled를 이용해 내가 지정한 style을 가지는 component를 새로 만들어 사용하는 방법. 한 component를 여러 스타일로 써야 하는데, 반복해서 사용해야 한다면 이렇게 사용하는게 코드를 읽기 쉬울 것 같다.
+
+3. **Higher-order component API**: withStyles을 사용해, Higher-order component가 내가 지정한 style을 가지는 component를 return하도록 하는 방법. 이건 왜 쓰는걸까... 모르겠다.
```
Comment (pawsong) [inline posts/8_materialUI2.md:85]: HoC 는 hooks 에 의해 대체된 흔적기관 같은 거... ㅎㅎ

https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750

HoC 가 각광을 받은 게 벌써 5년 전이네요... ㅎㅎ

**Comment by pawsong** [comment]
> @pawsong I added npm trends image as you commented. Check it plz! Plus, what do you think is better? Putting Material library's 2014 readme in Quote or Code? 

Code 가 나을 듯?

**Pull request merged**